src/screens/timeStamping/QRCodeScanner/index.tsx

```diff
@@ -0,0 +1,163 @@
+    default:
+      return state;
+  }
+};
+
+const QRCodeScanner = ({ route }: QRCodeScannerProps) => {
+  const [state, dispatch] = useReducer(reducer, { errorMessage: '', scanned: false, loading: false });
+  const [hasPermission, setHasPermission] = useState<boolean>(false);
+  const [modalVisible, setModalVisible] = useState<boolean>(false);
+
+  const navigation = useNavigation();
+
+  useFocusEffect(
+    useCallback(() => {
+      let isActive = true;
+
+      const requestPermission = async () => {
+        let { status } = await Camera.getPermissionsAsync();
+
+        if (isActive && status !== GRANTED) {
+          const { status: newStatus } = await Camera.requestPermissionsAsync();
+          status = newStatus;
+        }
+
+        if (isActive && status !== GRANTED) setModalVisible(true);
+
+        if (isActive) setHasPermission(status === GRANTED);
+      };
+
+      requestPermission();
+
+      return () => { isActive = false; };
+    }, [])
+  );
+
+  const handleBarCodeScanned = async ({ data }: BarCodeType) => {
+    try {
+      dispatch({ type: SCANNING });
+
+      if (data !== route.params.event.customer._id) {
+        dispatch({ type: WRONG_QR_CODE });
+        return;
+      }
+
+      const payload: timeStampEventPayloadType = { action: QR_CODE_TIME_STAMPING, startDate: new Date() };
+      await Events.timeStampEvent(route.params?.event?._id, payload);
+
+      goBack();
+    } catch (e) {
+      if ([409, 422].includes(e.response.status)) {
+        dispatch({ type: BAD_REQUEST, payload: { errorMessage: e.response.data.message } });
+      } else if ([404, 403].includes(e.response.status)) {
+        dispatch({ type: BAD_REQUEST, payload: { errorMessage: 'Vous ne pouvez pas horodater cet évènement.' } });
+      } else {
+        dispatch({
+          type: BAD_REQUEST,
+          payload: { errorMessage: 'Erreur, si le problème persiste, contactez le support technique.' },
+        });
+      }
+    }
+  };
+
+  const askPermissionAgain = async () => {
+    const permission = await Camera.requestPermissionsAsync();
+
+    if (!permission.canAskAgain) {
+      await Alert.alert(
+        'Accès refusé',
+        'Vérifiez que l\'application a bien l\'autorisation d\'accéder à l\'appareil photo.',
+        [{ text: 'OK', onPress: () => setModalVisible(false) }], { cancelable: false }
+      );
+    }
+
+    if (permission.status !== GRANTED) setModalVisible(true);
+
+    setHasPermission(permission.status === GRANTED);
+  };
+
+  const goBack = () => navigation.navigate('Home');
+
+  const goToManualTimeStamping = (eventStart: boolean) => {
+    navigation.navigate('ManualTimeStamping', { event: route.params.event, eventStart });
+  };
+
+  return (
+    <Camera onBarCodeScanned={!hasPermission || state.scanned || state.loading ? undefined : handleBarCodeScanned}
+      style={styles.container} barCodeScannerSettings={{ barCodeTypes: ['org.iso.QRCode'] }}>
```

Comment:
Sur mon telephone ca fonctionne avec le type conseillé par expo `BarCodeScanner.Constants.BarCodeType.qr`. Apres ca nous fait installer BarCodeScanner pas sure que ca vaille le coup, mais il faudrait peut etre voir si la constante n'existe pas ailleurs.

Reply:
La constante n'existe pas dans le package Camera, j'ai l'impression qu'il n'existe que dans BarCodeScanner.
En soit je peux supprimer cette vérification car on a notre propre gestion d'erreur. Je l'ai laissé car ça consomme moins de batterie de ne chercher qu'un seul type de code selon la doc expo

Reply:
J'ai juste peur que si un jour ils changent la valeur de la variable (meme si c'est peu probable), ca casse tout l'horodatage ^^'

Reply:
Je l'ai noté dans le ticket